tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala

```diff
@@ -341,7 +341,52 @@ class Quint(moduleData: QuintOutput) {
         case "setOfMaps" => binaryApp(opName, tla.funSet)
         case "set"       => ternaryApp(opName, tla.except)
         case "setBy"     => null
-        case "put"       => null
+        case "put"       =>
+          // put(map, key, value) ~~> (key :> value) @@ map
+          quintArgs =>
+            ternaryApp(opName,
+                (map, key, value) => {
+                  // (1) build `key :> value` == `[ _ \in {key} |-> value ]`
+                  // extract types
+                  val mapType = Quint.typeToTlaType(types(quintArgs(0).id).typ)
+                  val keyType = Quint.typeToTlaType(types(quintArgs(1).id).typ)
+                  val valType = Quint.typeToTlaType(types(quintArgs(2).id).typ)
+                  // f1 == key :> value
+                  val f1 = tla.funDef(value, (tla.name(uniqueVarName(), keyType), tla.enumSet(key)))
```

Comment:
Do we really want to go via this route? Functions are kind of expensive in the OOPSLA encoding, they are sets of pairs (though Jure's improvements may compensate for that).

If we directly specialize the definition of `@@` for the case of adding one pair, this should be more efficient and maybe even simpler.

Reply:
this looks to use the exact same approach as in our rewiring rule. why would we differ from that? or should that rule be changed?

Reply:
Ah, good point! See 435cb627e

Reply:
Oh i see! Cool :)

Reply:
Nice! :)